php/class-asset-manager-svg-sprite.php

```
@@ -0,0 +1,453 @@
+		$symbol_id        = $this->sprite_document->createAttribute( 'id' );
+		$symbol_id->value = $this->format_handle_as_symbol_id( $asset['handle'] );
+		$symbol->appendChild( $symbol_id );
+
+		// Use the viewBox attribute from the SVG asset.
+		$viewbox = $svg->getAttribute( 'viewBox' ) ?? '';
+
+		if ( ! empty( $viewbox ) ) {
+			$symbol_viewbox        = $this->sprite_document->createAttribute( 'viewBox' );
+			$symbol_viewbox->value = $viewbox;
+
+			$symbol->appendChild( $symbol_viewbox );
+		}
+
+		/* phpcs:disable WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase */
+
+		// Add the SVG's childNodes to the symbol.
+		foreach ( iterator_to_array( $svg->childNodes ) as $child_node ) {
+			if (
+				! ( $child_node instanceof DOMText ) // Exclude text nodes.
+				&& 'script' !== $child_node->nodeName // Exclude <script> tags.
+			) {
+				$symbol->appendChild( $this->sprite_document->importNode( $child_node, true ) );
+			}
+		}
+
+		/* phpcs:enable */
+
+		// Append the symbol to the SVG sprite.
+		$this->svg_root->appendChild( $symbol );
+
+		$this->asset_handles[]                = $asset['handle'];
+		$this->sprite_map[ $asset['handle'] ] = $asset;
+	}
+
+	/**
+	 * Returns the SVG markup for displaying a symbol.
+	 *
+	 * @param  string $handle The symbol handle.
+	 * @param  array  $attrs  Additional attributes to add to the <svg> element.
+	 * @return string         The <svg> and <use> elements for displaying a symbol.
+	 */
+	public function get_symbol( $handle, $attrs = [] ) {
+		if ( empty( $handle ) || ! in_array( $handle, array_keys( $this->sprite_map ), true ) ) {
+			return '';
+		}
+
+		$asset = $this->sprite_map[ $handle ];
+
+		if ( empty( $asset ) ) {
+			return '';
+		}
+
+		/*
+		 * Use the dimensions from `get_default_dimensions()` to calculate the
+		 * expected size when only one dimension is provided in $attrs.
+		 */
+		if ( ! empty( $asset['attributes']['width'] ) && ! empty( $asset['attributes']['height'] ) ) {
```

Again, should we return an error if we cannot parse the width and height?